lib/image_compare/modes/base.rb

```diff
@@ -36,9 +36,15 @@ def diff(bg, diff)
         diff.each do |pixels_pair|
           pixels_diff(diff_image, *pixels_pair)
         end
-        create_diff_image(bg, diff_image)
-          .highlight_rectangle(bounds)
-          .highlight_rectangle(include_rect, :green)
+
+        diff_image = create_diff_image(bg, diff_image)
+        if @different_areas.any?
```

Comment:
I think this validation is redundant (_the each method already ignores empty arrays by default_)